Input/GamepadTest/source/Gamepad.hx

```diff
@@ -202,72 +281,26 @@ class Gamepad extends FlxTypedGroup<FlxSprite>
 			sprite.visible = true;
 			sprite.x = FlxG.width * x;
 			sprite.y = FlxG.height * y;
-			sprite.x -= Std.int(sprite.width / 2);
-			sprite.y -= Std.int(sprite.height / 2);
 		}
 	}
 
-	function updateBar(bar:FlxBar, label:FlxText, isSupported:Bool, value:Float)
+	function updateBar(data:Bar, isSupported:Bool, value:Float)
 	{
-		bar.visible = label.visible = isSupported;
+		data.bar.visible = data.label.visible = isSupported;
 		if (isSupported)
-			bar.value = ((value + 1.0) / 2.0) * 100; // motion value range is from -1.0 to 1.0
-	}
-
-	function updateButton(button:FlxSprite, pressed:Bool)
-	{
-		button.alpha = pressed ? ALPHA_ON : ALPHA_OFF;
-	}
-
-	function updateButtonInvisible(button:FlxSprite, pressed:Bool)
-	{
-		button.alpha = pressed ? ALPHA_ON : ALPHA_INVISIBLE;
-	}
-
-	function updateShoulderButton(button:FlxSprite, pressed:Bool, pos:Int)
-	{
-		button.y = pos;
-		if (pressed)
-			button.y += 5;
-	}
-
-	function updateStick(stick:FlxSprite, xValue:Float, yValue:Float, pos:FlxPoint)
-	{
-		var angle:Float = 0;
-
-		if (xValue != 0 || yValue != 0)
-		{
-			angle = Math.atan2(yValue, xValue);
-			stick.x = pos.x + STICK_MOVEMENT_RANGE * Math.cos(angle);
-			stick.y = pos.y + STICK_MOVEMENT_RANGE * Math.sin(angle);
-			stick.alpha = ALPHA_ON;
-		}
-		else
-		{
-			stick.x = pos.x;
-			stick.y = pos.y;
-			stick.alpha = ALPHA_OFF;
-		}
-	}
-
-	function updateTrigger(yAxisValue:Float, sprite:FlxSprite, pos:Float)
-	{
-		yAxisValue = (yAxisValue + 1) / 2;
-		sprite.y = pos + (TRIGGER_MOVEMENT_RANGE * yAxisValue);
+			data.bar.value = ((value + 1.0) / 2.0) * 100; // motion value range is from -1.0 to 1.0
 	}
+}
 
-	function updateDpad()
-	{
-		var pressed = gamepad.pressed;
-
-		var left = pressed.DPAD_LEFT;
-		var right = pressed.DPAD_RIGHT;
-		var up = pressed.DPAD_UP;
-		var down = pressed.DPAD_DOWN;
+typedef Bar = { bar:FlxBar, label:FlxText };
```

Comment:
Good practice to make these types private if possible.

Reply:
The subtypes? I actually didn't know that was possible until I just googled it. I wish I knew this a long time ago